src/main/java/frc/robot/commands/AimAtSpeaker.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Constants.AutoConstants;
+import frc.robot.subsystems.Drive;
+
+public class AimAtSpeaker extends Command {
+  /** Creates a new AimAtSpeaker. */
+  private Drive drive;  
+  public AimAtSpeaker(Drive swerve) {
+    // Use addRequirements() here to declare subsystem dependencies.
+    this.drive = swerve;
+    addRequirements(drive);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    Pose2d difference = new Pose2d(AutoConstants.speakerPose.getX() -  drive.getPose().getX(), AutoConstants.speakerPose.getY() - drive.getPose().getY(), AutoConstants.speakerPose.getRotation().minus(drive.getRotation2d()));
```

Comment:
This shouldn't be a pose2d object. You should create a Translation2d object like: `speakerPose.getTranslation().minus(drive.getPose().getTranslation());` and then calculate the angle of that translation using Math.atan2()

Reply:
When you do that, it's also recommended that you modify the angle slightly to account for latency and motion. If your robot is moving and trying to aim at the target, it will be slow to respond to where the speaker is in relation to it. The robot's motion will also change the trajectory of the note when shooting. Both these things mean you need to increase or decrease the target angle to account for these factors and be able to shoot accurately. I wouldn't worry about the motion too much right now as it is more complex, but mainly the latency factor